File: tests/Generator/resource/python/oop/map.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, UserList, UserDict


P = TypeVar("P")
T = TypeVar("T")
class Map(BaseModel, Generic[P], Generic[T]):
    total_results: Optional[int] = Field(default=None, alias="totalResults")
    parent: Optional[P] = Field(default=None, alias="parent")
    entries: Optional[List[T]] = Field(default=None, alias="entries")
    pass

File: tests/Generator/resource/csharp/oop/Map.cs
using System.Text.Json.Serialization;

public class Map<P, T>
{
    [JsonPropertyName("totalResults")]
    public int? TotalResults { get; set; }

    [JsonPropertyName("parent")]
    public P? Parent { get; set; }

    [JsonPropertyName("entries")]
    public System.Collections.Generic.List<T>? Entries { get; set; }

}

